### PR TITLE
ENH: Port the new *anatomical fast-track* from *fMRIPrep*

### PR DESCRIPTION
**Pull request by oesteban**
Uses poldracklab/smriprep#107 to skip precalculated anatomical
derivatives, just setting the ``--anat-derivatives <path>`` command line
argument.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -163,17 +171,20 @@ jobs:
...
+          name: Get anatomical derivatives for ds001771
           command: |
-            if [[ ! -d /tmp/data/ds001771/derivatives/freesurfer-6.0.1 ]]; then
+            if [[ ! -d /tmp/data/ds001771/derivatives ]]; then
               mkdir -p /tmp/data/ds001771/derivatives
               wget --retry-connrefused --waitretry=5 --read-timeout=20 --timeout=15 -t 0 -q \
-                -O ds001771_sub-36_derivatives_freesurfer.tar.gz "https://files.osf.io/v1/resources/8k95s/providers/osfstorage/5e21e74fedceab00ad82e047"
-              tar xvzf ds001771_sub-36_derivatives_freesurfer.tar.gz -C /tmp/data/ds001771/derivatives
-              mv /tmp/data/ds001771/derivatives/freesurfer /tmp/data/ds001771/derivatives/freesurfer-6.0.1
+                -O ds001771-deriv.tar.gz "https://files.osf.io/v1/resources/8k95s/providers/osfstorage/5eb251fba2199500215d1dc3"
```
Comment (josephmje) [inline .circleci/config.yml:179]:
```suggestion
                -O ds001771-derivs.tar.gz "https://files.osf.io/v1/resources/8k95s/providers/osfstorage/5eb251fba2199500215d1dc3"
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -163,17 +171,20 @@ jobs:
...
+              tar xvzf ds001771-deriv.tar.gz -C /tmp/data/ds001771/
```
Comment (josephmje) [inline .circleci/config.yml:180]:
```suggestion
              tar xvzf ds001771-derivs.tar.gz -C /tmp/data/ds001771/
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -390,8 +367,10 @@ jobs:
                 -v /tmp/ds001771/work:/work \
                 --user $(id -u):$(id -g) \
                 nipreps/dmriprep:latest /data /out participant -vv \
-                --fs-subjects-dir /data/derivatives/freesurfer-6.0.1 --sloppy \
-                --notrack --skip-bids-validation -w /work --omp-nthreads 2 --nprocs 2
+                -w /work --omp-nthreads 2 --nprocs 2 \
+                --notrack --skip-bids-validation --sloppy \
+                --fs-subjects-dir /data/derivatives/freesurfer-6.0.1 \
```
Comment (josephmje) [inline .circleci/config.yml:372]: I'm a little bit confused about specifying both `--fs-subjects-dir` and `--anat-derivatives`. Is the idea to have only one of these arguments exist in the future?

Comment (oesteban) [reply .circleci/config.yml:372]: It's not, in principle. These two flags give the flexibility to run FS in isolation and then connect it to further anatomical processing, again in isolation. Assuming that both go together brings us back to the original situation where we need to have freesurfer, smriprep and dmriprep folders under the output folder, if one wants to reuse cached results.

**Pull request merged**